src/ark/segmentation/ez_seg/ez_seg_utils.py

```diff
@@ -0,0 +1,134 @@
+    """
+    for fov in tqdm(io_utils.list_folders(tiff_dir, substrs=fovs)):
+        shutil.copytree(os.path.join(tiff_dir, fov), dst=os.path.join(mantis_dir, fov))
+
+        for seg_type in io_utils.list_folders(segmentation_dir):
+
+            for mask in io_utils.list_files(os.path.join(segmentation_dir, seg_type), substrs=fov):
+                shutil.copy(os.path.join(segmentation_dir, seg_type, mask),
+                            dst=os.path.join(mantis_dir, fov)
+                            )
+
+
+def log_creator(variables_to_log: dict, base_dir: str, log_name: str = "config_values.txt"):
+    # Define the filename for the text file
+    output_file = os.path.join(base_dir, log_name)
+
+    # Open the file in write mode and write the variable values
+    with open(output_file, "w") as file:
+        for variable_name, variable_value in variables_to_log.items():
+            file.write(f"{variable_name}: {variable_value}\n")
+
+    print(f"Values saved to {output_file}")
+
+
+def filter_csvs_by_mask(csv_path_name: Union[str, pathlib.Path], csv_name: str) -> None:
+    """
+    Function to take in and separate a single cell table into multiple based on the mask_type parameter.
+    Args:
+        csv_path_name (Union[str, pathlib.Path]):
+            The path to the directory containing the raw image data.
+        csv_name (str):
+            The path to the directory containing the raw image data.
+    """
+    # Load the CSV file as a DataFrame (replace 'input.csv' with your CSV file)
+
+    for item in io_utils.list_files(csv_path_name):
+        input_csv_file = os.path.join(csv_path_name, item)
+        df = pd.read_csv(input_csv_file)
+
+        # Define the column to filter
+        column_to_filter = 'mask_type'  # Replace with the actual column name
+
+        # Get unique values from the specified column
+        filter_values = df[column_to_filter].unique()
+
+        # Create a dictionary to store filtered DataFrames
+        filtered_dfs = {}
+
+        # Filter the DataFrame for each unique value and save as separate CSV files
+        for filter_value in filter_values:
+            filtered_df = df[df[column_to_filter] == filter_value]
+
+            # Define the output CSV file name based on the filtered value
+            table_type_str = item.replace(csv_name, '')
+            output_csv_file = os.path.join(csv_path_name, ''.join([f'filtered_{filter_value}', table_type_str]))
+
+            # Save the filtered DataFrame to a new CSV file
+            filtered_df.to_csv(output_csv_file, index=False)
+
+            # Store the filtered DataFrame in the dictionary
+            filtered_dfs[filter_value] = filtered_df
+
+        # Print a message for each filtered DataFrame
+        #for filter_value, filtered_df in filtered_dfs.items():
+            #print(f"Filtered DataFrame for '{filter_value}' saved as filtered_{filter_value}.csv")
```

Comment:
If this isn't going to be printed out anymore we should remove.

Reply:
removed